### ISSUE TITLE
Error when running Notebook

### ISSUE THREAD

**Issue by fscorrupt** [jfeil/HelloFresh-Tandoor-Converter#1]
Possible that the Code is broken?
Cant get it to work :/

Error i get:

`---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [2], in <cell line: 43>()
     38             self._request_new_token()
     40         return self._access_token
---> 43 default_bearer = _Bearer()
     46 def request_recipes(offset=0, limit=250):
     47     headers = {'Authorization': f'Bearer {str(default_bearer)}'}

Input In [2], in _Bearer.__init__(self)
     17 def __init__(self):
     18     self._access_token = ""
---> 19     self._request_new_token()

Input In [2], in _Bearer._request_new_token(self)
     22 answer = requests.get('https://www.hellofresh.com/recipes/')
     23 self._cookies = answer.cookies
---> 24 token_answer = re.search("{tokenType.*}}", answer.text).group()
     25 return_dict = {}
     26 for element in token_answer.split(','):

AttributeError: 'NoneType' object has no attribute 'group'`

**Comment by jfeil** [comment]
Yes, I just checked it. They changed their way of generating a new bearer token. You can get one manually via the Inspect feature of Firefox/Chrome. I will try to fix it also for others in the future :)

Just set the default_bearer in the second cell to the received string (e.g. default_bearer = ...)

![grafik](https://user-images.githubusercontent.com/11638228/157111922-7253f891-4b52-4a48-a393-242dca1a34d6.png)

**Comment by fscorrupt** [comment]
Thanks for the fast fix, sadly there is another error at the end of the script.


![image](https://user-images.githubusercontent.com/45659314/157181326-544ad778-6b75-4acd-b5fb-efaae6b6b665.png)

**Comment by jfeil** [comment]
This is more likely due to a missing requirements.txt ^^
You probably installed progressbar, but the module that is used, should be progressbar2. I cannot confirm this until the end of the day, but it would be worth a try :)

**Comment by fscorrupt** [comment]
Yeah you where right! after installing progressbar2 it works, Thanks for the great work on this one.!

**Comment by jfeil** [comment]
Perfect! I might add a requirements.txt for others in the future 😅

**Comment by seh-len** [comment]
Does the converter still work? I have never used jupyter before, so the problem may be on my end. 
I am getting the error message “JSONDecodeError: Expecting value: line 1 column 1 (char 0)”.

**Comment by jfeil** [comment]
Hey, I have not checked it since the last fix, so I cannot say if this still works. But will check it for you in the next days :)

**Comment by seh-len** [comment]
Hello Jan,

I have identified several issues that need to be addressed. First, Cloudflare does not allow simple requests.post, so it is better to switch to scraper = cloudscraper.create_scraper().
Another problem with my tandoor was that keyword_ans = requests.get(f'{base_url}/api/keyword/?query={element[“id”]}', headers=headers) always returns a result, but the keyword you're looking for isn't in first place, and an existing keyword is returned, which means the creation fails.
However, I then noticed that HelloFresh now has so many duplicates of the same recipe in its database with only minimal differences that I didn't want to fill my tandoor with them and changed my mind, deciding that I would rather import the recipes individually.

Thank you anyway for the inspiration.

[HelloFresh2Tandoor.txt](https://github.com/user-attachments/files/20541792/HelloFresh2Tandoor.txt)
